src/analysis/convex/basic.lean

```diff
@@ -582,32 +588,62 @@ calc
     add_le_add (mul_le_mul_of_nonneg_left (le_max_left _ _) ha) (mul_le_mul_of_nonneg_left (le_max_right _ _) hb)
   ... ≤ max (f x) (f y) : by rw [←add_mul, hab, one_mul]
 
-lemma convex_on.le_on_segment (hf : convex_on s f) {x y z : E}
+lemma convex_on.le_on_segment {f : E → ℝ} (hf : convex_on s f) {x y z : E}
   (hx : x ∈ s) (hy : y ∈ s) (hz : z ∈ [x, y]) :
   f z ≤ max (f x) (f y) :=
```

Comment:
Can this not be generalized to `β` because `β` has no `max`?

Reply:
Yeah. The other option would be to generalize it to a β with a linear order (and a semimodule with ℝ as the ring), so basically pretty much just ℝ...

Reply:
Maybe add a doc-string to this effect?

Reply:
> The other option would be to generalize it to a β with a linear order (and a semimodule with ℝ as the ring), so basically pretty much just ℝ...

Hmm, what if I defined a lexicographic ordering on pairs `β = (ℝ, ℝ)`, and gave them the obvious semimodule structure.

This lemma would then hold even though `β` is not `ℝ`, right?

Reply:
Done!

Reply:
Oops -- sorry @eric-wieser, I think you must have added this comment at the same time as I confirmed the merger. I think it's probably simpler now to keep this in mind for the next PR on the subject: I'll be defining `concave_on` next, so I'll have to make another pass on these lemmas anyway.

Reply:
Yeah, not a big deal - its not like you actively removed a generalization, you just missed one that could be added :)